rxfeedback/src/main/java/org/notests/rxfeedback/Feedbacks.kt

```diff
@@ -204,6 +164,186 @@ fun <State, Query, Event> reactSetSafe(
         }
 
 
+/**
+ * The purpose of QueryLifetimeTracking is to activate, deactivate,
+ * as well as update already activated effects with the new Query that was returned.
+ */
+private class QueryLifetimeTracking<Query, QueryID, Event>(
+        val effects: (intialQuery: Query, state: Observable<Query>) -> Observable<Event>,
```

Comment:
Now the `effects` accepts the `state: Observable<Query>` as a parameter, but in all our current helpers we're just ignoring this parameter: 

```kotlin
effects = { initial: Query, _ ->
            effects(initial)
}
```

What is the use case of this `state` parameter?

Reply:
The idea of the general form is to have some standard way to implement nested systems while still updating the inner systems with the parent substate/state changes.

The `react` and `reactSet` can be seen as a projection (subset) of the general form. It would be better to provide the general form but also refactor existing `react` methods to use the general form in order to reduce the code base...

Reply:
Yeah, I understand why we have a general form. What I meant is that the general form is more general then all our current use cases (actually, my use cases :)). I thought that the implementation in swift was created with a specific use case in mind and wanted to know what that use case is. @kzaher did you use this general form for something that cannot be described without a general form?